src/creature_states_prisn.c

```diff
@@ -162,7 +164,7 @@ short creature_drop_body_in_prison(struct Thing *thing)
 
 }
 
-struct Thing *find_prisoner_for_thing(struct Thing *creatng)
+struct Thing *find_prisoner_for_thing(struct Thing *creatng, SpellKind spell_idx)
```

Comment:
hmm maybe the function name could use an update, for_thing seems vague

Reply:
The function already excludes frozen units, so I suggest naming it 'find_prisoner_to_freeze', and you could comment on it too that it returns a unit in any prison that is not frozen.